general/mem_manager.hpp

```diff
@@ -105,17 +141,17 @@ class Memory
    friend class MemoryManager;
    friend void MemoryPrintFlags(unsigned flags);
 
-   enum FlagMask
+   enum FlagMask: unsigned long
```

Comment:
Why make this `unsigned long`? The actual `flag` in class `Memory` is just `unsigned`. I guess the right type should be `unsigned` instead of the previous `int` or the current `unsigned long`.

Reply:
You're right, I can't see the warning I remember I wanted to avoid. `reverted`